### PR TITLE
feat: [PIE-5139]: add ability to enable FF via .env

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/global.d.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@
 declare const __DEV__: boolean
 declare const Bugsnag: any
 declare const __BUGSNAG_RELEASE_VERSION__: string
+declare const DEV_FF: Record<string, boolean>
```
Comment (abhinavrastogi-harness) [inline src/global.d.ts:13]: this assumes that value will be boolean. i think thats correct, but @udayvunnam mentioned some flags are not using boolean values. can we confirm with him once.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10891)

**Review by udayvunnam — APPROVED** [review]

File: src/global.d.ts (hunk 1 above)
Comment (vkbansal-harness) [reply src/global.d.ts:13]: We do not use anything other than `booleans`. Even `FeatureFlagMap` is of the type `Partial<Record<FeatureFlag, boolean>>`

**Comment by vkbansal-harness** [comment]
retrigger titlecheck

**Comment by vkbansal-harness** [comment]
retrigger jest

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**